Simplenote/SharedStorageMigrator.swift

```diff
@@ -0,0 +1,154 @@
+    }()
+
+    static let groupDocumentsDirectory: URL? = {
+        guard let groupDirectory = groupDirectory else {
+            return nil
+        }
+        return groupDirectory.appendingPathComponent(Constants.documentDirectory)
+    }()
+
+    static let oldDbURL: URL? = {
+        guard let documentsURL = documentsURL else {
+            return nil
+        }
+
+        return documentsURL.appendingPathComponent(Constants.sqlFile)
+    }()
+
+    static let newDbURL: URL? = {
+        guard let groupDocumentsDirectory = groupDocumentsDirectory else {
+            return nil
+        }
+        return groupDocumentsDirectory.appendingPathComponent(Constants.sqlFile)
+    }()
+
+    /// Database Migration
+    /// To be able to share data with app extensions, the CoreData database needs to be migrated to an app group
+    /// Must run before Simperium is setup
+    static func migrateCoreDataToAppGroup() {
+        guard let oldDbURL = oldDbURL,
+              let groupDocumentsDirectory = groupDocumentsDirectory,
+              let newDbURL = newDbURL else {
+            return
+        }
+
+        // Testing prints
+        // TODO: Remove prints later
+        print("oldDb exists \(FileManager.default.fileExists(atPath: oldDbURL.path))")
+        print(oldDbURL.path)
+        print("newDb exists \(FileManager.default.fileExists(atPath: newDbURL.path))")
+        print(newDbURL.path)
+
+        let persistentStoreCoordinator = NSPersistentStoreCoordinator(managedObjectModel: SPAppDelegate.shared().managedObjectModel)
+
+        if FileManager.default.fileExists(atPath: newDbURL.path) {
+            // Migration previously completed
+            NSLog("Database migration not needed")
+
+            // exit
+            return
+        }
+
+        if !FileManager.default.fileExists(atPath: oldDbURL.path) && !FileManager.default.fileExists(atPath: newDbURL.path) {
+            NSLog("New database needed")
+            NSLog("Creating database in app group")
+
+            // No DB exists
+            // Create new DB in shared group
+            createAppGroupDirectory(at: groupDocumentsDirectory)
+            addPersistentStore(to: persistentStoreCoordinator, at: newDbURL)
+
+            // Exit
+            return
+        }
+
+        if FileManager.default.fileExists(atPath: oldDbURL.path) && !FileManager.default.fileExists(atPath: newDbURL.path) {
+            // Old DB.  Needs Mirgation
+            NSLog("Database needs migration to app group")
+            NSLog("Beginning database migration")
+
+            createAppGroupDirectory(at: groupDocumentsDirectory)
+
+            // Option 1: Migrated old DB to new location
+            migrateDatabase(from: oldDbURL, to: newDbURL, coordinator: persistentStoreCoordinator)
+
+            // Option 2: Migrate old DB FILES to new location
+//            migrateCoreDataFiles(from: documentsURL, to: groupDocumemntsDirectory)
```

Comment:
IMO this is a much cleaner approach. We'd wanna avoid spawning all of the NSMO entities